### PR TITLE
Add Arabic Speech Corpus 

### PR DESCRIPTION
**Pull request by zaidalyafeai**

### CODE REVIEW

File: datasets/arabic_speech_corpus/arabic_speech_corpus.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            for line in f:
+                wav_file, phonetic = line.split('"')[1::2]
+                phonetics[wav_file] = phonetic
+
+        with open(orthographic_path, "r", encoding="utf-8") as f:
+            for line in f:
+                wav_file, orthographic = line.split('"')[1::2]
+                orthographics[wav_file] = orthographic
+
+        for _id, lab_name in enumerate(os.listdir(lab_dir)):
```
Comment (lhoestq) [inline datasets/arabic_speech_corpus/arabic_speech_corpus.py:129]:
```suggestion
        for _id, lab_name in enumerate(sorted(os.listdir(lab_dir))):
```
we need to make the order deterministic here

**Review by lhoestq — COMMENTED** [review]
Awesome !! Thanks for adding this one :)

And thanks for the example using `soundfile`, this is helpful.

This looks all good to me !

For audio datasets we try to have only 1 example in the dummy data to not make the repository too big. Before we merge, could you try to keep only 1 example ? (`ARA NORM  0003.wav` for example, since it's the smallest one ?)

File: datasets/arabic_speech_corpus/arabic_speech_corpus.py (hunk 1 above)
Comment (zaidalyafeai) [reply datasets/arabic_speech_corpus/arabic_speech_corpus.py]: Thanks @lhoestq , made the changes.

File: datasets/arabic_speech_corpus/README.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+### Languages
+
+The audio is in Arabic.
+
+## Dataset Structure
+
+### Data Instances
+
+A typical data point comprises the path to the audio file, usually called `file` and its transcription, called `text`.
+
```
Comment (lhoestq) [inline datasets/arabic_speech_corpus/README.md:73]:
````suggestion
An example from the dataset is:
```
{
    'file': '/Users/username/.cache/huggingface/datasets/downloads/extracted/baebe85e2cb67579f6f88e7117a87888c1ace390f4f14cb6c3e585c517ad9db0/arabic-speech-corpus/wav/ARA NORM  0002.wav',
    'orthographic': 'waraj~aHa Alt~aqoriyru Al~a*iy >aEad~ahu maEohadu >aboHaA^i haDabapi Alt~ibiti fiy Alo>akaAdiymiy~api AlS~iyniy~api liloEuluwmi - >ano tasotamir~a darajaAtu AloHaraArapi wamusotawayaAtu Alr~uTuwbapi fiy Alo<irotifaAEi TawaAla ha*aA Aloqarono',
    'phonetic': "sil w a r a' jj A H a tt A q r ii0' r u0 ll a * i0 < a E a' dd a h u0 m a' E h a d u0 < a b H aa' ^ i0 h A D A' b a t i0 tt i1' b t i0 f i0 l < a k aa d ii0 m ii0' y a t i0 SS II0 n ii0' y a t i0 l u0 l E u0 l uu0' m i0 sil < a' n t a s t a m i0' rr a d a r a j aa' t u0 l H a r aa' r a t i0 w a m u0 s t a w a y aa' t u0 rr U0 T UU0' b a t i0 f i0 l Ah i0 r t i0 f aa' E i0 T A' w A l a h aa' * a l q A' r n sil",
    'text': '\ufeffwaraj~aHa Alt~aqoriyru Al~aTHiy >aEad~ahu maEohadu >aboHaA^i haDabapi Alt~ibiti fiy Alo>akaAdiymiy~api AlS~iyniy~api liloEuluwmi - >ano tasotamir~a darajaAtu AloHaraArapi wamusotawayaAtu Alr~uTuwbapi fiy Alo<irotifaAEi TawaAla haTHaA Aloqarono'
}
```
````

**Review by lhoestq — APPROVED** [review]
Thank you :)

I also added one example from the dataset in the dataset card

**Pull request merged**